src/imitation/rewards/reward_nets.py

```diff
@@ -82,20 +83,29 @@ def preprocess(
         del state, action, next_state, done  # unused
 
         # preprocess
-        state_th = preprocessing.preprocess_obs(
-            state_th,
-            self.observation_space,
-            self.normalize_images,
+        state_th = cast(
```

Comment:
It might be worth adding an SB3 PR to allow type-safe polymorphism through generics.

Reply:
I mean, to be fair, we aren't checking anywhere in the class that the spaces we are given are array spaces. Maybe we should do that? Otherwise it seems unwarranted to have the cast.

Reply:
Oh damn, this is fixed in the new gym release (they made Space a generic), but we aren't using it due to SB3.